concrete/controllers/single_page/login.php

```diff
@@ -388,6 +391,20 @@ public function logout($token = false)
         }
     }
 
+    /**
+     * @param $token
+     * @return RedirectResponse
+     */
+    public function do_logout($token = false)
```

Comment:
Yeah, the old method exits so imagine this code:

```php
if (!$this->testUserForSomething($user)) {
    // uh oh! Our test failed, lets log the user out to be safe
    $this->logout($token->generate('logout'));
}

// Now empty the bank accounts for the user since they passed the test
$this->giveAllMoneyAndAssetsAlsoFriendshipsAndFamilyKeepSakes($user);
```